### ISSUE TITLE
mydumper broken with slave replication

### ISSUE THREAD

**Issue by discostur** [maxbube/mydumper#236]
I compiled latest mydumper from github master branch. I'm trying to dump mysql databases on a mysql slave:

```
** (mydumper:19953): CRITICAL **: 15:26:07.680: There are queries in PROCESSLIST running longer than 60s, aborting dump,
	use --long-query-guard to change the guard value, kill queries (--kill-long-queries) or use 
	different server for dump
``` 

I think it has problems with the replication process running:

```
mysql> show processlist;
+----+-----------------+-----------+------+---------+---------+--------------------------------------------------------+------------------+
| Id | User            | Host      | db   | Command | Time    | State                                                  | Info             |
+----+-----------------+-----------+------+---------+---------+--------------------------------------------------------+------------------+
|  4 | event_scheduler | localhost | NULL | Daemon  | 2761948 | Waiting on empty queue                                 | NULL             |
| 15 | system user     |           | NULL | Connect | 2761455 | Waiting for master to send event                       | NULL             |
| 16 | system user     |           | NULL | Query   | 2719764 | Slave has read all relay log; waiting for more updates | NULL             |
| 50 | root            | localhost | NULL | Query   |       0 | starting                                               | show processlist |
+----+-----------------+-----------+------+---------+---------+--------------------------------------------------------+------------------+
``` 

If i stop mysql replication slave then i can dump DBs without errors.

**Comment by Andor** [comment]
@discostur try `--trx-consistency-only` option.

**Comment by discostur** [comment]
@Andor i applied the option but it does not help:

```
** (mydumper:62858): WARNING **: 13:53:30.666: Using trx_consistency_only, binlog coordinates will not be accurate if you are writing to non transactional tables.

** (mydumper:62858): CRITICAL **: 13:53:30.671: There are queries in PROCESSLIST running longer than 60s, aborting dump,
	use --long-query-guard to change the guard value, kill queries (--kill-long-queries) or use 
```

**Comment by liuqian1990** [comment]
--kill-long-queries

**Comment by maxbube** [comment]
Hi @discostur I will review this once I have some time, maybe we just need to exclude `system user` or also Commands `Daemon` and `Connect` as we do for query.

**Comment by maxbube** [comment]
In the meantime as workaround you can just set `--long-query-guard` bigger than the Time you get in PROCESSLIST

**Comment by discostur** [comment]
@maxbube that would be nice - just say me if i should test anything

**Comment by jeremymaes** [comment]
@davidducos are there any plans on fixing this one (perhaps with the fix Dupondje made) in a release anytime soon?
We were hoping this would have made the cut for 0.10 but unfortunately it seems that it didn't.

As we have a lot of master-slave or master-master setups being upgraded to MySQL 8.0 we are quite badly impacted by this. Using the workaround maxbube suggested works, but it is somewhat impractical as you need to use a ridiculously high value (since the system threads "Time" keeps increasing as long as you don't restart MySQL). It also defeats the purpose of the built-in safety.

**Comment by davidducos** [comment]
@jeremymaes 
0.10.1 was my first release, I didn't add any new pull request or feature as it was my first release and I had to get used to with the release procedure. 
I have the intention of release version 0.10.3 on March, and this is one of the issues that I want to fix that I want to include.
I'm going to work with @maxbube for the approval.
